sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala

```diff
@@ -322,15 +335,23 @@ private[testkit] class TopicImpl(
   override def publish(message: ByteString, metadata: SdkMetadata): Unit =
     brokerProbe.emit(message, metadata)
 
-  override def publish[T <: GeneratedMessageV3](message: TestKitMessage[T]): Unit =
-    publish(message.getPayload.toByteString, message.getMetadata)
+  override def publish(message: TestKitMessage[?]): Unit = message.getPayload match {
+    case javaPb: GeneratedMessageV3 => publish(javaPb.toByteString, message.getMetadata)
+    case scalaPb: GeneratedMessage  => publish(scalaPb.toByteString, message.getMetadata)
+    case str: String                => publish(ByteString.copyFromUtf8(str), message.getMetadata)
+    case _ =>
+      val encodedMsg = JsonSupport.getObjectMapper
+        .writerFor(message.getPayload.getClass)
+        .writeValueAsBytes(message.getPayload)
+      publish(ByteString.copyFrom(encodedMsg), message.getMetadata)
```

Comment:
Same comment as above, we should use `JsonSupport.encodeJson` instead. We must understand why this is not working.

Reply:
Similarly to the other occurrence but reversed, here we are offering to serialize the object for the user and in a real scenario where the user is writing to a topic, the payload will not be preceded by the special proto tag that "encodeJson" includes, so I don't think we should use it here at all. Also, note that the user is not forced to use our serialization, it can use `publish(ByteString, Metadata)` and push whatever bytes they want directly.

Reply:
This seams reasonable, given that `publish(Array[Byte])` puts exactly those bytes in the pretend topic, and presents those bytes to eventing in the proxy, just like if they came from a message broker. The proxy will then TLA-encode the bytes for passing around and possibly hand those bytes to an action for example.

Same should go in the other direction, the proxy should decode `Any` with our custom type_url into non-prefixed bytes when publishing to a topic.

There might be something slightly related to attaching metadata with content type `application/json` when auto-transforming like this and just publishing the raw bytes like if to a broker topic.

Reply:
I see that metadata magic is in the TestkitMessageImpl